src/jit/decomposelongs.cpp

```diff
@@ -1056,24 +1055,22 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
                 }
                 else
                 {
-                    Range().Remove(loOp1);
-
                     assert(count >= 32);
 
-                    // Zero out loResult (shift of >= 32 bits shifts all lo bits to hiResult)
-                    loResult = m_compiler->gtNewZeroConNode(TYP_INT);
-                    Range().InsertBefore(tree, loResult);
-
                     if (count < 64)
                     {
                         if (count == 32)
                         {
                             // Move loOp1 into hiResult (shift of 32 bits is just a mov of lo to hi)
-                            hiResult = loOp1;
-                            Range().InsertBefore(tree, hiResult);
+                            LIR::Use loOp1Use(Range(), &gtLong->gtOp.gtOp1, gtLong);
+                            loOp1Use.ReplaceWithLclVar(m_compiler, m_blockWeight);
+
+                            hiResult = loOp1Use.Def();
+                            Range().Remove(gtLong);
                         }
                         else
                         {
+                            Range().Remove(gtLong);
```

Comment:
We aren't actually seeing that behavior though on the >32 case. In the > 32 case, it appears we generate the hi op before the lo op:

N001 (  3,  2) [000007] ------------        t7 =    lclVar    int    V07 rat4          $140

               [000025] ------------       t25 =    const     int    1

                                                 /--*  t7     int    
                                                 +--*  t25    int    
               [000026] ------------       t26 = *  <<        int   

               [000027] ------------       t27 =    const     int    0

                                                 /--*  t27    int    
N005 ( 15, 13) [000011] DA----------             *  st.lclVar int    V07 rat4         

IN0007: 000014 8BD0         mov      edx, eax
IN0008: 000016 D1E2          shl      edx, 1
IN0009: 000018 33C0         xor      eax, eax